target/linux/ath79/image/generic.mk

```diff
@@ -999,6 +999,21 @@ define Device/librerouter_librerouter-v1
 endef
 TARGET_DEVICES += librerouter_librerouter-v1
 
+define Device/meraki_mr16
+  SOC := ar7161
+  DEVICE_VENDOR := Meraki
+  DEVICE_MODEL := MR16
+  ROOTFS_SIZE := 13440k
+  IMAGE_SIZE := 15680k
+  DEVICE_PACKAGES := kmod-spi-gpio kmod-owl-loader
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-rootfs | pad-rootfs
+  IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | pad-to $$$$(ROOTFS_SIZE) | append-kernel | append-metadata | check-size
+  IMAGES := kernel.bin rootfs.bin sysupgrade.bin
+  SUPPORTED_DEVICES += meraki_mr16
```

Comment:
Just "mr16", that's the string from ar71xx.

Reply:
I see. That may explain the `sysupgrade` refusal.

Reply:
Fixed in 8d699c6.

Reply:
Depending on how we solve the MAC issue, we might drop the SUPPORTED_DEVICES eventually, so leave this open as a reminder.

Reply:
I should undo my fix in 8d699c6 so this check fails on upgrading from ar71xx.

`platform_check_image` should fail here, because right now the upgrade step described in the [most recent revision](https://github.com/openwrt/openwrt/commit/359b1ed3f1dac316f4223bd1a4368034951d1d38) demands the user's direct intervention to
- (step 5) set the kernel address in u-boot config, and
- (step 6) to modify `/lib/upgrade/comon.sh` to pad with 128K zeroes to fix alignment.

The latter we could solve with a second target image only for upgrading, with a 128K zero pad to the image, but the former we cannot fix without user intervention (or doing something very bad like extending the firmware partition into the u-boot config region).

Let me know what you think.

Reply:
Have a look at what the new DEVICE_COMPAT_VERSION makes possible here:
https://github.com/openwrt/openwrt/pull/3345/commits/06336990c55d356abb30d4f4dac1dab2c002eff7#diff-0835d68ee1baa61ee8893cec8a75f77eR49

Reply:
OK. Updated and built, and, as expected, my MR16 flashed with previous build rejects it due to compat_version being unset.

After sysupgrading -F it, it does like it, and `uci show system.@system[0].compat_version` is 2.0.